src/containers/Welcome/index.js

```diff
@@ -14,40 +20,100 @@ const BottomAnchoredBackgroundImage = () => (
   </RootView>
 );
 
-export default class Welcome extends React.Component {
-  static propTypes = {
-    navigation: PropTypes.shape({
-      navigate: PropTypes.func.isRequired,
-    }).isRequired,
+const BrandingIcon = () => (
+  <Image
+    style={{
+      alignSelf: 'flex-end',
+      marginRight: 44,
+      marginTop: 42,
+    }}
+    source={MojoTechIcon}
+  />
+);
+
+class Welcome extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.nameInput = null;
+
+    this.state = {
+      guestName: '',
+      timer: setInterval(this.tick, 1000),
```

Comment:
I think this might be a change request: We're currently causing a re-render every second. What if we instead do something like this:

```javascript
const timeoutSeconds = 60;

constructor(props) {
  ..snip...
  this.restartTimeout();
}

restartTimeout = () => {
  if (this.timer) clearInterval(this.timer);
  this.timer = setInterval(this.handleTimeout, timeoutSeconds * 1000);
}

handleTimeout = () => {
  if (!this.nameInput.isFocused()) return;
  Keyboard.dismiss();
  this.nameInput.blur();
  this.setState({guestName: ''});
}

handleChangeNameText = text => this.setState({ guestName: text }, this.restartTimeout);
```

(Also handles the off-by-one issue ;) )